### PR TITLE
Add support for health checks

### PR DESCRIPTION
**Pull request by danielwitz**

### CODE REVIEW

File: gbus/abstractions.go [hunk 1]
```diff
@@ -149,6 +158,10 @@ type Builder interface {
 
 	//WithPolicies defines the default policies that are applied for evey outgoing amqp messge
 	WithPolicies(policies ...MessagePolicy) Builder
+
+	//WithDbPingTimeout defines the default timeout in seconds for the db ping check
+	WithDbPingTimeout(timeoutInSeconds time.Duration) Builder
```
Comment (rhinof) [inline gbus/abstractions.go]: Let's change this to ConfigureHealthCheck as I wouldn't want to add additional methods if we want to evolve the ability to configure the health check later on the road.

File: gbus/builder/builder.go [hunk 2]
```diff
@@ -149,11 +157,16 @@ func (builder *defaultBuilder) Txnl(provider, connStr string) gbus.Builder {
 }
 
 func (builder *defaultBuilder) WithSerializer(serializer gbus.MessageEncoding) gbus.Builder {
-
 	builder.serializer = serializer
 	return builder
 }
 
+func (builder *defaultBuilder) WithDbPingTimeout(timeoutInSeconds time.Duration) gbus.Builder {
```
Comment (rhinof) [inline gbus/builder/builder.go]: should be changed  to reflect the comment I add to the interface def

File: gbus/bus.go [hunk 3]
```diff
@@ -281,6 +283,23 @@ func (b *DefaultBus) NotifyHealth(health chan error) {
 	b.healthChan = health
 }
 
+//GetHealth implements Health.GetHealth
+func (b *DefaultBus) GetHealth() HealthCard {
+	dbInitiated := b.TxProvider != nil
```
Comment (rhinof) [inline gbus/bus.go]: instead please check b.IsTxnl

File: gbus/tx/provider.go [hunk 4]
```diff
@@ -3,22 +3,38 @@ package tx
 import (
 	"context"
 	"database/sql"
+	"time"
 )
 
-//Provider for PostgreSQL
+var ctx context.Context
```
Comment (rhinof) [inline gbus/tx/provider.go]: why do we need this declaration ?

File: gbus/bus.go [hunk 5]
```diff
@@ -281,6 +283,23 @@ func (b *DefaultBus) NotifyHealth(health chan error) {
...
+	var dbConnected bool
+
+	if dbConnected = false; dbInitiated {
```
Comment (rhinof) [inline gbus/bus.go]: wouldn't dbConnected always evaluate to false here ? if so please remove from the if check

File: gbus/bus.go [hunk 6]
```diff
@@ -281,6 +283,23 @@ func (b *DefaultBus) NotifyHealth(health chan error) {
...
+		dbConnected = b.TxProvider.Ping(b.DbPingTimeout)
+	}
+
+	return HealthCard{
+		DbInitiated:        dbInitiated,
```
Comment (rhinof) [inline gbus/bus.go]: Not sure this should be on the HealthCard what does DbInitiated mean ?

File: gbus/bus.go (hunk 5 above)
Comment (danielwitz) [reply gbus/bus.go]: it's  a short way to write this:
```suggestion
	if dbInitiated {
	dbConnected = true 
	} else{
	dbConnected = false
	}
```
I'll change it to be more readable

File: gbus/bus.go (hunk 6 above)
Comment (danielwitz) [reply gbus/bus.go]: from outside I want to know if to check the DbConnected value in the health check. If the grabbit is not transactional and only uses rabbit the health check should reflect that.

Comment (danielwitz) [reply gbus/bus.go]: Is `UsingDb`  or `IsTransactional` better?

Comment (rhinof) [reply gbus/bus.go]: once the bus is configured to be transactional then it uses a DB and the health check should include checking the DB. so it is enough for a caller to configure the bus as transactional and for the health check to check whether IsTxnl

**Pull request merged**